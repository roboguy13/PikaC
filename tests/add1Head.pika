%generate add1Head

Sll : layout[x](List);
Sll Nil := emp;
Sll (Cons head tail) :=
  x :-> head **
  (x+1) :-> nxt **
  Sll tail [nxt];

add1Head : Sll -> Sll;
add1Head Nil := Nil [Sll];
add1Head (Cons h t) :=
  Cons (h + 1) t [Sll];

