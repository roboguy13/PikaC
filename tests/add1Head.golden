*** Running test add1Head1
(2 50462800 (2 50462784 (3 0 ())))

